tests/unit/data_formats/test_yolo_format.py

```diff
@@ -77,42 +76,67 @@ def randint(a, b):
 
 class CompareDatasetMixin:
     @pytest.fixture(autouse=True)
-    def setup(self, helper_tc):
+    def setup(self, helper_tc, monkeypatch):
         self.helper_tc = helper_tc
+        self.monkeypatch = monkeypatch
 
     def compare_datasets(self, expected, actual, **kwargs):
         compare_datasets(self.helper_tc, expected, actual, **kwargs)
 
 
 class CompareDatasetsRotationMixin(CompareDatasetMixin):
     def compare_datasets(self, expected, actual, **kwargs):
-        actual_copy = copy.deepcopy(actual)
-        compare_datasets(self.helper_tc, expected, actual, ignored_attrs=["rotation"], **kwargs)
-        for item_a, item_b in zip(expected, actual_copy):
-            for ann_a, ann_b in zip(item_a.annotations, item_b.annotations):
-                assert ("rotation" in ann_a.attributes) == ("rotation" in ann_b.attributes)
-                assert (
-                    abs(ann_a.attributes.get("rotation", 0) - ann_b.attributes.get("rotation", 0))
-                    < 0.01
+        def compare_rotated_annotations(expected: Bbox, actual: Bbox, ignored_attrs=None):
+            if expected.type != AnnotationType.bbox or actual.type != AnnotationType.bbox:
+                return compare_annotations(expected, actual, ignored_attrs=ignored_attrs)
+
+            ignored_attrs = (ignored_attrs or []) + ["rotation"]
+            rotation_diff = expected.attributes.get("rotation", 0) - actual.attributes.get(
+                "rotation", 0
+            )
+            rotation_diff %= 180
+            rotation_diff = min(rotation_diff, 180 - rotation_diff)
+            assert rotation_diff < 0.01 or abs(rotation_diff - 90) < 0.01
+            if rotation_diff < 0.01:
+                return compare_annotations(expected, actual, ignored_attrs=ignored_attrs)
+            if abs(rotation_diff - 90) < 0.01:
+                x, y, w, h = actual.get_bbox()
+                center_x = x + w / 2
+                center_y = y + h / 2
+                new_width = h
+                new_height = w
+                actual = Bbox(
+                    x=center_x - new_width / 2,
+                    y=center_y - new_height / 2,
+                    w=new_width,
+                    h=new_height,
+                    label=actual.label,
+                    attributes=actual.attributes,
                 )
+                return compare_annotations(expected, actual, ignored_attrs=ignored_attrs)
+
+        self.monkeypatch.setattr(
+            "datumaro.util.test_utils.compare_annotations", compare_rotated_annotations
+        )
```

Comment:
This is unnecessarily hacky, IMO. You could add a parameter to `compare_dataset` to override the comparison function to use for annotations.

Reply:
added a parameter